### PR TITLE
REF: GH38174 - Refactoring DataFrame.combine_first

### LINKED ISSUE #38174: REF/CLN: DataFrame.combine_first
xref https://github.com/pandas-dev/pandas/pull/38145#issuecomment-735346546

`DataFrame.combine_first` should be refactored to use `BlockManager.where` instead of `pandas.core.computation.expressions.where`

### PR DESCRIPTION
**Pull request by hydratedguy**
- [ ] closes #38174 
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).

### CODE REVIEW

**Comment by hydratedguy** [comment]
Am i using the right where function? Do i have to use self._mgr.where(...) or self.where(...) ?

File: pandas/core/frame.py [hunk 1]
```diff
@@ -7822,20 +7822,20 @@ def combine_first(self, other: DataFrame) -> DataFrame:
...
-            x_values = extract_array(x, extract_numpy=True)
             y_values = extract_array(y, extract_numpy=True)
 
             # If the column y in other DataFrame is not in first DataFrame,
             # just return y_values.
             if y.name not in self.columns:
                 return y_values
 
-            return expressions.where(mask, y_values, x_values)
+            values = self._mgr.where(y_values, mask, align=True, axis=1)
```
[the remark this answers is not in the document]
Comment (hydratedguy) [reply pandas/core/frame.py:7836]: Actually it's breaking because of the axis parameter that shouldn't be there, but even without it still break tests

**Comment by phofl** [comment]
@hydratedguy closing as stale. Please ping of you want to continue

**Comment by hydratedguy** [comment]
Hey @phofl ! From what i've seen i think that to solve [#38174](https://github.com/pandas-dev/pandas/issues/38174) it's needed to do changes on a Internals file, so there's not much i can do. What do you think?

**Comment by phofl** [comment]
You can fix this through making changes in the internals or use a different function

**Comment by hydratedguy** [comment]
But only a pandas dev can do changes in the internals right? And if i used a different function i woulnd't be following what the PR asked for.

**Comment by phofl** [comment]
This is the pandas dev repo. Every contributor can make changes

**Comment by hydratedguy** [comment]
Hmm got it. Even so, I believe that as a beginner I am not able to solve this problem rn. Sorry for bothering you and thanks for answering me!

**Pull request closed without merging**